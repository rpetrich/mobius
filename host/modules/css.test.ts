import { packageRelative } from "../fileUtils";
import css from "./css";
import { VirtualModule } from "./index";

describe("integration test", () => {

	describe("css/styles.css", () => {
		const virtualModule = css(packageRelative("tests/css"), packageRelative("tests/css/styles.css"), false, () => {}) as VirtualModule;

		describe("generated type declaration", () => {
			it("should match expected source", () => {
				expect(virtualModule.generateTypeDeclaration()).toMatchSnapshot();
			});
		});

		describe("generated module", () => {
			it("should match expected source", () => {
				expect(virtualModule.generateModule()).toMatchSnapshot();
			});
		});

		describe("generated styles", () => {
			const styles = virtualModule.generateStyles!(["styled"]);
			it("should match expected source", () => {
				expect(styles.css).toMatchSnapshot();
			});
		});

	});

});
