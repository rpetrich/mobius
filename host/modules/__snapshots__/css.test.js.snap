// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integration test css/styles.css generated module should match expected source 1`] = `
"export const styled = \\"_tests_css_styles_styled\\";
export const unused = \\"_tests_css_styles_unused\\";"
`;

exports[`integration test css/styles.css generated styles should match expected source 1`] = `
"._tests_css_styles_styled {
	background: red;
	width: 42px;
}
"
`;

exports[`integration test css/styles.css generated type declaration should match expected source 1`] = `
"export const styled: string;
export const unused: string;"
`;
